frontend/test/pytest/test_autograph.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""PyTests for the AutoGraph source-to-source transformation feature."""
+
+import numpy as np
+import pennylane as qml
+import pytest
+
+import catalyst
+from catalyst import qjit
+from catalyst.ag_primitives import STD
+from catalyst.autograph import _TRANSFORMER as AGT
+
+# pylint: disable=missing-function-docstring
+
+
+class TestIntegration:
```

Comment:
Should we have any integration tests that contain if statements?

Reply:
💯 Am going to add those :)

Reply:
https://github.com/PennyLaneAI/catalyst/pull/235/commits/b63ed16d91730ac73a4f7145d2f117d5f9048035